docs: guide for porting an html site to gatsby

Jeddf opened the pull request:
<!--
  Have any questions? Check out the contributing docs at https://gatsby.dev/contribute, or
  ask in this Pull Request and a Gatsby maintainer will be happy to help :)
-->

<!--
  Is this a blog post? Check out the docs at https://www.gatsbyjs.org/contributing/blog-and-website-contributions/, and please mention if the blog post is pre-approved
  by someone from Gatsby.
-->

## Description

This PR adds a [reference guide](https://www.gatsbyjs.org/contributing/docs-templates/#reference-guides) for porting part of a HTML website to Gatsby.

For context, and possible further use, I've pushed up the whole example site and finished port to a [personal repo](https://github.com/Jeddf/gatsby-partial-html-materials).

Possible improvements:

- This PR places it in the [Porting to Gatsby](https://www.gatsbyjs.org/docs/porting-to-gatsby/)  reference guide overview but it feels closer to a [tutorial](https://www.gatsbyjs.org/contributing/docs-templates/#tutorials) as it strives to stay focused for an audience that has a very particular task in mind and may only have worked with HTML and CSS. Might be better broken up into tutorials or with some parts replaced by links to recipes?

- Remove S3 asset links? (I think they're handy for demonstrating the utility of assetPrefix but probably not suitable for inclusion in the guide)
- Create a basic starter with Helmet and a basic Layout component already added so we can skip those parts?

## Related Issues

Fixes #13813

## Preview (in 2 parts 😂) - updated for 02061ccb

![Page Preview Part 1](https://user-images.githubusercontent.com/2471488/68325523-8186ba80-00c1-11ea-8b7d-fad2e6bee4be.jpg)
![Page Preview Part 2](https://user-images.githubusercontent.com/2471488/68325570-995e3e80-00c1-11ea-9a1d-5a63c5f72ac3.jpg)

gillkyle commented on docs/docs/porting-part-of-a-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,532 @@
+
+### Path Prefix
+
+By default, Gatsby assumes that the build will be hosted at the root path of a domain, e.g `https://gatsbyjs.org/`. After the initial static HTML file is loaded, Gatsby reaches out for other generated JavaScript files in the same location in order to progressively enable enhancements for the user. In the example site, the path at which the Gatsby site will be hosted is `/who`. The gatsby `pathPre ...
```
```suggestion
By default, Gatsby assumes that the build will be hosted at the root path of a domain, e.g `https://gatsbyjs.org/`. After the initial static HTML file is loaded, Gatsby reaches out for other generated JavaScript files in the same location in order to [progressively enable enhancements](/docs/glossary/#progressive-enhancement) for the user. In the example site, the path at which the Gatsby site will be hosted is `/who`. The gatsby `pathPrefix` option will ensure that the generated JavaSript files are available to load and the generated links from `index.js` to internal pages work correctly:
```

Jeddf commented on docs/docs/porting-part-of-a-html-site-to-gatsby.md:
````diff
@@ -0,0 +1,532 @@
+export default () => <div>Hello world!</div>
+```
+
+Run the development server in the command line to see the website:
````
Yeh I agree, I have a strong urge to keep 'em on the page but this is a good spot to remind them about the rest of the docs that they can move onto after too.

Jeddf commented:
Thanks @gillkyle, appreciate that coming from the CRA port master! ⚓️

Yeh I kept looking at spots to link to in the docs but they didn't quite seem to fit with trying to minimize JavaScript and stick to the basic, static application. Definitely think there might be other approaches though, it's an interesting one.

muescha commented:
I miss in the further steps section:

- how I port a second/third folder to Gatsby? There is only one pathPrefix: Do i need a second, third Gatsby for this?
- how I port the full site to Gatsby? (By add who folder to the src director, removing pathPrefix ...)

Jeddf commented:
Thanks for taking a look @muescha! Those ideas for further steps both sound good, I'll look at adding them in later today. 👍.
I'm not sure about a neat solution for porting multiple sections, will do a little thinking/reading on that.

Jeddf commented:
> I miss in the further steps section:
> 
> * how I port a second/third folder to Gatsby? There is only one pathPrefix: Do i need a second, third Gatsby for this?
> * how I port the full site to Gatsby? (By add who folder to the src director, removing pathPrefix ...)

Done! They sorta popped out as one section together, think it rounds the guide off nicely, really appreciate the suggestion. Interested to hear your thoughts on it @muescha. I also added a bit about Gatsby handling assets, the next steps section is maybe looking a little heavy now!

muescha commented on docs/docs/porting-part-of-a-html-site-to-gatsby.md:
````diff
@@ -521,11 +521,48 @@ website-domain
+    └── shrinking.js
+```
+
+2. The `pathPrefix` configuration option is no longer needed as the Gatsby application will now be served at the root of the site alongside the remaining HTML files.
````
but we still need the build with  `--withPaths` for the assets?

Jeddf replied:
Aye, there's a note up with the build step on line 479 that I think does the trick in explaining this:

`> **Note**: The `--prefix-paths` option _must_ be used for path and asset prefixes to be applied`

Thinking they'll find themselves back up there when building again if they hit problems.

marcysutton replied:
Here's a question: what does the asset prefix stuff gain? Could this guide be simplified and include global CSS instead?

Jeddf replied:
Looking again at the [asset prefix docs](https://www.gatsbyjs.org/docs/asset-prefix/) page as well, I think it'll follow on from this guide really cleanly for the reader. I'll take a pass through to take out asset-prefix from the example, host the whole gatsby build output at the /who path and move the css, png etc to assets folder on the website-domain.

Jeddf replied:
What do you mean by global CSS? Directly importing a CSS file in the Layout.js file? I'd lean more toward keeping a basic link tag so the method stays consistent with the remaining HTML pages, so

`<link
      rel="stylesheet"
      type="text/css"
      href="https://gatsby-html-partial-assets.s3.eu-west-2.amazonaws.com/style.css"
    />`

becomes

`<link rel="stylesheet" type="text/css" href="/assets/style.css"/>`

muescha commented on docs/docs/porting-part-of-a-html-site-to-gatsby.md:
```diff
@@ -521,11 +521,48 @@ website-domain
+
+2. Adjust the `pathPrefix` Gatsby configuration option to reflect the new root path at which the Gatsby application will be served.
+
+3. Ensure the `to` props on all Gatsby `<Link>` components are correct relative to `/src/pages/` as the application root.
```
general question (maybe adressed also early on this page): here we use `<Link>` for all pages in the old and new website or only for the new website (preloading is compatible with old html pages?)

muescha replied:
i think you adressed this in the next point 3

Jeddf replied:
The [docs](https://www.gatsbyjs.org/docs/gatsby-link/#replace-a-tags-with-the-link-tag-for-local-links) warn against using `<Link>` for external sites, I took that to mean anything outside the Gatsby app but looking back it may just be external to the website as a whole.

Once `pathPrefix` is not being used in a project, I don't know if there is a reason not to use `<Link>` for non Gatsby, internal pages. That is a possible small clarification to add to the [Link page](https://www.gatsbyjs.org/docs/gatsby-link/#replace-a-tags-with-the-link-tag-for-local-links). Will dig a little to find the answer.

muescha commented on docs/docs/porting-part-of-a-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,569 @@
+
+### Integrated site file structure
+
+Here is the structure of the HTML & non JavaScript asset files after the built, Gatsby `/who` section is added to website-domain and the `/gatsby/who/` folder is added at the asset domain:
```
```suggestion
Here is the structure of the HTML & non JavaScript asset files after the built, Gatsby `/who` section is added to website domain and the `/gatsby/who/` folder is added at the asset domain:
```

marcysutton commented on docs/docs/porting-part-of-a-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,569 @@
+
+### Path Prefix
+
+The `pathPrefix` option in `/gatsby-config.js` tells Gatsby the path at which the build output will be served from. Links to other pages and to the generated JavaScript files that provide [progressive enhancements](/docs/glossary/#progressive-enhancement) for the user will be prefixed with the `pathPrefix` value. This is the config addition for the example Gatsby application, served at the `/who` ...
```
What do you mean by "generated JavaScript files" here? Isn't that covered by pages? "other pages" could also refer to the pages that haven't been touched by Gatsby, and if they were prefixed with `/who`, might not be correct.

```suggestion
The `pathPrefix` option in `/gatsby-config.js` tells Gatsby the path at which the build output will be served from. For example, links to other pages within your Gatsby site will be prefixed with the `pathPrefix` value. Here is some config code to tell Gatsby it should only handle and care about the `/who` path for now:
```

Jeddf replied:
I was trying to acknowledge the many new JavaScript files that we are asking them to host. My thinking is that the audience here will be used to just having a collection of known HTML and CSS files to host. Thinking now that it should be addressed directly with a section if at all though. If assetPrefix is not part of the example I think it's less important to address.

Will pick this up in the discussion below where you brought up assetPrefix.

marcysutton commented on docs/docs/porting-part-of-a-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,569 @@
+
+### Asset Prefix
+
+The `assetPrefix` option tells Gatsby about an external location, the asset domain, where its compiled JavaScript files will be served from. In addition to the `pathPrefix`, the link paths for Gatsby's generated JavaScript files will be prefixed with the `assetPrefix` value if provided. In the example website there are three assets already hosted at an asset domain; the style sheets `style.css` a ...
```
This seems like overkill to me -- I think it makes more sense to introduce global CSS so people can edit it directly, and we can skip the complexity for asset prefixes (unless it's needed for something specific related to porting within a project, like a pathPrefix is).

marcysutton commented on docs/docs/porting-part-of-a-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,569 @@
+
+### Build step
+
+You now have a site that mirrors the existing HTML site section. Stop the development server if it's still running, it's time to run the build! 🎉
```
```suggestion
You now have a site that mirrors the existing HTML site section. Stop the development server if it's still running; it's time to run the production build! 🎉
```

marcysutton commented on docs/docs/porting-part-of-a-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,569 @@
+
+### Integrated site file structure
+
+Here is the structure of the HTML & non JavaScript asset files after the built, Gatsby `/who` section is added to website domain and the `/gatsby/who/` folder is added at the asset domain:
```
```suggestion
Here is the structure of the HTML & non-JavaScript asset files after the built Gatsby `/who` section is added to website domain and the `/gatsby/who/` folder is added at the asset domain:
```

marcysutton commented on docs/docs/porting-part-of-a-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,569 @@
+
+2. The `pathPrefix` configuration option is no longer needed as the Gatsby application will now be served at the root of the site alongside the remaining HTML files.
+
+3. Add `/who` to the start of the `to` attributes in the existing index.js, and do the equivalent for `/services`. In the `Layout.js` component file, change the `<a>` tag linking to the services page to a Gatsby `<Link>` component.
```
A short code example would be helpful here.

marcysutton commented on docs/docs/porting-part-of-a-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,569 @@
+
+## Next steps
+
+Gatsby can handle assets through direct imports to page and component files, the [asset import documentation](/docs/importing-assets-into-files/) covers imports as well as the less optimized static folder. Once assets are handled through Gatsby, plugins can be used to optimize their processing and delivery.
```
```suggestion
Gatsby can handle assets through direct imports to page and component files; the [asset import documentation](/docs/importing-assets-into-files/) covers imports as well as the less-optimized `static` folder. Once assets are handled through Gatsby, plugins can be used to optimize their processing and delivery.
```

marcysutton commented on docs/docs/porting-part-of-a-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,569 @@
+
+Gatsby can handle assets through direct imports to page and component files, the [asset import documentation](/docs/importing-assets-into-files/) covers imports as well as the less optimized static folder. Once assets are handled through Gatsby, plugins can be used to optimize their processing and delivery.
+
+The [building with components page](/docs/building-with-components/) has information about why Gatsby uses React component architecture and how it fits into a Gatsby application.
```
```suggestion
The [building with components doc](/docs/building-with-components/) has information about why Gatsby uses React component architecture and how it fits into a Gatsby application.
```

marcysutton commented on docs/docs/porting-part-of-a-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,569 @@
+
+[Sourcing content and data](/docs/content-and-data/) is a great next step if you are interested in separating your content from your website code.
+
+Short guides can be found at the [recipes section](/docs/recipes), for adding functionality such as optimizing and querying local images, adding markdown support and integrating various modern CSS tools. The [adding website functionality page](/docs/adding-website-functionality/) has longer guides for larger tasks such as making your site accessible, adding authentication and doing client side da ...
```
```suggestion
Short guides can be found at the [recipes section](/docs/recipes) for adding functionality such as optimizing and querying local images, adding Markdown support and integrating various modern CSS tools. The [adding website functionality page](/docs/adding-website-functionality/) has longer guides for larger tasks such as making your site accessible, adding authentication, and fetching data with client-side JavaScript.
```

Jeddf commented on docs/docs/porting-part-of-a-html-site-to-gatsby.md:
````diff
@@ -0,0 +1,569 @@
+cd gatsby-site-section
+```
+
+The `/src` folder contains all code for the Gatsby site. In the Gatsby build stage [every file in the `/src/pages` folder will result in a HTML file](/docs/recipes/#creating-pages-automatically). Currently, the only page file is the root `/src/pages/index.js`:
````
Ah yes, seeing an interesting skill I need to work on here: [progressively disclose complexity](https://www.gatsbyjs.org/docs/gatsby-core-philosophy/#progressively-disclose-complexity) while not being misleading.

Jeddf commented on docs/docs/porting-part-of-a-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,569 @@
+  </head>
+  <body>
+    <header>
+      <h3 class="brand-color">Taylor's Tidy Trees</h3>
```
Aha! Yes. This is really a logo and Deque uni says [best practice is to start main content with a h1](https://dequeuniversity.com/assets/html/jquery-summit/html5/slides/headings.html#best_practice_start_content_with_h1) So I'll styling this logo with a class and get a h1 on every page. Looking back the pages do all have a heading shaped gap.

Jeddf commented on docs/docs/porting-part-of-a-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,569 @@
+ * See: https://www.gatsbyjs.org/docs/gatsby-config/
+ */
+
+module.exports = {
```
Great, yeh wanted to keep it clear of GraphQL, agree with giving it a direct mention at the end.

Jeddf commented on docs/docs/porting-part-of-a-html-site-to-gatsby.md:
````diff
@@ -0,0 +1,569 @@
+  </>
+)
+```
+
````
Grand, I went back and forth on mentioning JSX directly, agree with bringing it in. Certainly in danger of misleading the reader about what javascript is if it's not mentioned.

Jeddf commented on docs/docs/porting-part-of-a-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,569 @@
+
+### Gatsby Project
+
+Now that you are setup we'll be using the Gatsby and NPM CLI tools on your command line to get this site section ported!
```
Ack! Was sure I had internalized that rule, should have done a quick cmd-f.

Jeddf commented on docs/docs/porting-part-of-a-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,569 @@
+
+### Development environment
+
+Gatsby generates websites and compiles applications, but also has a tool for local development. With some setup your computer can run generation and the local development environment. Follow the [Gatsby development environment tutorial](/tutorial/part-zero/) to get setup and then return here to continue.
```
Agreed! It implied that generation and compilation are different but wasn't clear how/why. Your version clears up that confusion without disclosing too much complexity.

Jeddf commented on docs/docs/porting-part-of-a-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,569 @@
+
+## Why Gatsby?
+
+### Web building blocks
```
Very much agree, I've tried to help the reader focus in the rest of the page but this section would certainly make my mind wander. Pushing up a rewrite shortly, keen to hear your thoughts on it.

Jeddf commented on docs/docs/porting-part-of-a-html-site-to-gatsby.md:
````diff
@@ -0,0 +1,569 @@
+      └── sam-surgeon.html
+```
+
+The `/who` section of the site is a great candidate for porting as it is all within a single folder. Through this guide, you will develop the ported Gatsby section in isolation before integrating into the site.
````
Agreed, pushing up this addition shortly.

Jeddf commented:
Wow, it's fantastic to have my work be the focus of such a thorough, thoughtful review, thankyou very much. I've finished implementing your suggestions and replying to comments for now.

I really like the idea of it ultimately being a proper reference guide that someone can work through in bitesize chunks. I'll definitely keep on thinking about how to move it in that direction.

muescha commented:
Since Marcy asked maybe we can avoid the assetprefix ([comment](https://github.com/gatsbyjs/gatsby/pull/18609#discussion_r336704706)) I think also about this:

Is it possible for simplified guide to avoid the `pathPrefix`, if the guide from beginning rebuild the site structure under `src/pages` so there is no need to explain the `pathPrefix` In this guide.

Jeddf commented:
Yeh! Now that the page is about porting HTML in general, tangling it up with path prefix all the way through just possibly confuses.

Maybe there are 2 topics here really. Migrating an HTML site to gatsby and nesting a Gatsby app into an HTML site. Can flip it to a complete porting first then talk about the prefix stuff and it'll be easier to link out at that point.

I'll extract path prefix as a next iteration. Then take another look at what can be linked out/moved to recipes/other pages I reckon the change will make lotsa other pages more applicable.

Jeddf commented:
I've restructured it to be a direct porting from HTML guide 🎉 Screenshot is not updated unfortunately. Will take a pass tomorrow over it with the gatsby docs, to see what I think can be linked out/moved to recipes/other pages.

muescha commented:
Looks very good
Maybe you also mention the `pathPrefix` in the last section if someone like more strong encapsulated parts?

Jeddf commented:
Yeh I think enough should be done in the page for a user to be able to locate the prefix instructions at a logical spot and understand the use case. Added a hosting section with enough direction to apply pathPrefix while also linking out to the docs pages for path and asset prefix. What do you think?

Jeddf commented:
Happy Halloweeeeen @marcysutton! I've addressed your suggestions and made an additional structural change to suit it now not just being a partial porting guide as suggested by @muescha. The core example is now a full HTML site migration, and pathPrefix is offered as an optional step around build and deploy if the site is not to be hosted at the domain root.

The opening positions it relative to the main tutorial. Despite it being somewhat similar, I think there is still significant value in providing a shorter, focused path for new users to get from HTML to a Gatsby static site.

It could go full reference guide by hedging closer to the format of the CRA porting page and addressing each of the points somewhat in isolation rather than structuring it around an overall example. I think in that case though you'd ask a little more of the reader by having them need to thread together an end to end development flow themselves. I'm assuming the HTML porting user's current build process will likely just be edit file and upload/commit. While the CRA structure suits addressing points likely to be hit when adapting their existing workflow, the HTML user is likely to be starting a development flow from scratch, so structuring around an example and walkthrough makes sense.

🎃

marcysutton commented on docs/docs/porting-an-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,543 @@
+
+In the `/static` folder of your new Gatsby project, you can see a `favicon.ico` file, this is the Gatsby favicon! All files within `/static` will be served at the root of the app, you can put yours in there and watch it show up in the browser. The other image file in this example, `person.png`, should also be moved to the `/static` folder for later use.
+
+Gatsby negates the need for CSS `<link>` tags for local CSS. Within the `/src/pages/index.js` file, before the page component is defined, you can add a line for importing each CSS file. Gatsby will then efficiently deliver the CSS with your site.
```
```suggestion
Gatsby's bundling system negates the need for manual CSS `<link>` tags for local CSS. Within the `/src/pages/index.js` file, before the page component is defined, you can add a line for importing each CSS file. Gatsby will then efficiently deliver the CSS with your site.
```

marcysutton commented on docs/docs/porting-an-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,543 @@
+
+Gatsby negates the need for CSS `<link>` tags for local CSS. Within the `/src/pages/index.js` file, before the page component is defined, you can add a line for importing each CSS file. Gatsby will then efficiently deliver the CSS with your site.
+
+Create a folder at `/src/styles` in the Gatsby project. All the css files for the project can be moved into this new folder. Add an import line for each css file to the Gatsby home page:
```
```suggestion
Create a folder at `/src/styles` in the Gatsby project. All the CSS files for the project can be moved into this new folder. Add an `import` line for each CSS file to the Gatsby home page:
```

marcysutton commented on docs/docs/porting-an-html-site-to-gatsby.md:
````diff
@@ -0,0 +1,543 @@
+
+Now you can import the `<Helmet>` component to the `index.js` file and place `<header>` & `<main>` elements for the existing HTML. Copy over the contents of the `<head>` tag, You'll no longer need the `<link>` tags for the CSS files. The Gatsby components must have a single root parent in their code structure so one technique is to add a [React Fragment component](https://reactjs.org/docs/fragmen ...
+
+```jsx:title=/src/pages/index.js
````
Should this be `/gatsby-site/src/pages/index.js`? Same as below.

marcysutton commented on docs/docs/porting-an-html-site-to-gatsby.md:
````diff
@@ -0,0 +1,543 @@
+}
+```
+
+Now you can import the `<Helmet>` component to the `index.js` file and place `<header>` & `<main>` elements for the existing HTML. Copy over the contents of the `<head>` tag, You'll no longer need the `<link>` tags for the CSS files. The Gatsby components must have a single root parent in their code structure so one technique is to add a [React Fragment component](https://reactjs.org/docs/fragmen ...
````
```suggestion
Now you can import the `<Helmet>` component to the `index.js` file and place `<header>` & `<main>` elements for the existing HTML. Copy over the contents of the `<head>` tag: you'll no longer need the `<link>` tags for the CSS files. The Gatsby components must have a single root parent in their code structure so one technique is to add a [React Fragment component](https://reactjs.org/docs/fragments.html) around them:
```

marcysutton commented on docs/docs/porting-an-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,543 @@
+
+Opening the site in a browser again at `http://localhost:8000`, you should have a visually complete home page! You'll make the links functional next by porting more pages. Before that, this guide will briefly explore how HTML and JavaScript combine as JSX in a Gatsby application.
+
+### HTML and JavaScript
```
This small word change evokes a bit more detail:

```suggestion
### HTML from JavaScript
```

marcysutton commented on docs/docs/porting-an-html-site-to-gatsby.md:
````diff
@@ -0,0 +1,543 @@
+)
+```
+
+Have a check that the `Who We Are` link is still working.
````
If it isn't, do you have any advice for what they should do next to investigate?

Jeddf replied:
Good point, I think the only way it'd go wrong at that point is if elements the Layout is replacing haven't been completely removed:
```suggestion
Have a check that the `Who We Are` link is still working. If not, check all the elements being replaced by the `<Layout>` component has been removed.
```

That might be clear enough with the example above though, maybe the line could just be deleted. What do you think?

Jeddf replied:
Decided to updated with a note to check with the code snippet directly above the line:

>Have a check that the `Who We Are` link is still working. If not, check that the content is wrapped correctly with the `<Layout>` component as shown above.

marcysutton commented on docs/docs/porting-an-html-site-to-gatsby.md:
````diff
@@ -0,0 +1,543 @@
+Once `/services` and the root level HTML files are ported, here is what the finished Gatsby project file structure looks like:
+
+```
+gatsby-site
````
Love this! I think it will be very helpful.

marcysutton commented on docs/docs/porting-an-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,543 @@
+
+### Gatsby build step
+
+You now have a site that mirrors the existing HTML site section. Stop the development server if it's still running; it's time to run the production build! 🎉
```
```suggestion
With all of the pages ported over, you now have a site that mirrors the existing HTML site. Stop the development server if it's still running; it's time to run the production build! 🎉
```

In case they missed the note about porting over `/services`, this gives them another nudge to catch up. What do you think?

marcysutton commented on docs/docs/porting-an-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,543 @@
+
+### Hosting the new website
+
+Once built, the contents of the `/public` folder are ready to be hosted at the root (`/`) of a domain. The files can be deployed in exactly the same way that your existing HTML site was deployed. For more deployment options, see the [Deploying and Hosting page](/docs/deploying-and-hosting/).
```
I don't think we'll know exactly how their site was deployed, and dragging the public folder into FTP can be a bit of a pain. Perhaps we could make it more open-ended?
```suggestion
Once built, the contents of the `/public` folder are ready to be hosted at the root (`/`) of a domain. The files can be deployed in similar ways to how your existing HTML site may have been deployed. For more deployment options including automation with git and cloud services, see the [Deploying and Hosting page](/docs/deploying-and-hosting/).
```

marcysutton commented on docs/docs/porting-an-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,543 @@
+
+If the Gatsby site is to be hosted at a non-root path, e.g. `example.com/blog/`, Gatsby needs to be informed so page and asset links in the built output can be prefixed.
+
+[Path prefix](/docs/path-prefix/) is the Gatsby option for setting a non-root hosting path. Here is the example project `gatsby-config.js` file with a path prefix added for `/blog`:
```
This is nice! 👍

marcysutton commented on docs/docs/porting-an-html-site-to-gatsby.md:
````diff
@@ -0,0 +1,543 @@
+```shell
+gatsby build --prefix-paths
+```
+
````
What do you think of including a tip about gatsby-serve? I could see this tripping people up otherwise.
````suggestion

You can also serve the built site locally for review using the `prefix-paths` flag:

```shell
gatsby serve --prefix-paths
```
````

Jeddf replied:
Excellent! I didn't know about serve 👍

marcysutton commented on docs/docs/porting-an-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,543 @@
+
+## Next steps
+
+Gatsby can handle images through direct imports to page and component files too! The [asset import documentation](/docs/importing-assets-into-files/) covers this. Once assets are handled through Gatsby, plugins can be used to optimize their processing and delivery.
```
```suggestion
Gatsby can handle images through direct imports to page and component files too! The [asset import documentation](/docs/importing-assets-into-files/) covers this. There is also the [Gatsby Image](/docs/gatsby-image/) component for even deeper optimizations. Once assets are handled through Gatsby, plugins can be used to optimize their processing and delivery.
```
I feel like we should mention Gatsby Image here too...

marcysutton commented on docs/docs/porting-an-html-site-to-gatsby.md:
```diff
@@ -0,0 +1,543 @@
+
+The [building with components doc](/docs/building-with-components/) has information about why Gatsby uses React component architecture and how it fits into a Gatsby application.
+
+[Sourcing content and data](/docs/content-and-data/) is a great next step if you are interested in separating your content from your website code, such as sourcing the site title from `gatsby-config.js` with GraphQL.
```
```suggestion
[Sourcing content and data](/docs/content-and-data/) is a great next step if you are interested in separating your content from your website code, such as sourcing the site title from `gatsby-config.js` with GraphQL and writing content in Markdown.
```

muescha commented:
#13813 requests some incremental change. This is lost now In the rewriting. 

Maybe at some point we can stop in the middle (maybe after changing the about section) and show how to deploy an half migrated site? 

Maybe to answer the questiin: Can the files from `/public` mixed with the source files on the server?

marcysutton commented:
> #13813 requests some incremental change. This is lost now In the rewriting.
> Maybe at some point we can stop in the middle (maybe after changing the `who` section) and show how to deploy a half migrated site?

That's a good point. @Jeddf I'd recommend addressing it at the beginning of the doc to help the reader establish whether they want to read the whole thing by talking specifically about those two use cases. At the end of the intro paragraph, you could say something like "This doc could also be applied to migrate a section of a site to be served next to existing files; pay extra attention to the section on Hosting..."

> Maybe to answer the question: Can the files from `/public` mixed with the source files on the server?

That could be addressed in the Building & Deploying section, near where you talk about hosting and the `--path-prefix`.
